homeassistant/helpers/enum.py

```diff
@@ -0,0 +1,20 @@
+"""Enum related helpers."""
+
+from enum import Enum
+from typing import Any
+
+
+class StrEnum(str, Enum):
+    """Partial backport of Python 3.11's StrEnum for our basic use cases."""
+
+    def __new__(cls, value: str, *args: Any, **kwargs: Any) -> "StrEnum":
+        """Create a new StrEnum instance."""
+        # enum.auto intentionally not backported yet, waiting until it's
```

Comment:
I think we should implement the generator method for `auto()` and raise if it is used.
This prevents it from being used, especially when we are at the point of supporting both Python 3.10 & 3.11.

Reply:
We used to raise already as a side effect of not accepting anything but a `str`:
```
>>> class MyEnum(StrEnum):
...  FOO = auto()
TypeError: 1 is not a string
```
425df3b30bf4d2bd0ea23d5f26116128dca7c5a2 makes it more explicit. Ran into a mypy/pylint issue while doing that, hence the pylint disable.